pkg/docker/config/config.go

```diff
@@ -75,17 +83,23 @@ func SetAuthentication(sys *types.SystemContext, registry, username, password st
 // GetAuthentication returns the registry credentials stored in
 // either auth.json file or .docker/config.json
 // If an entry is not found empty strings are returned for the username and password
-func GetAuthentication(sys *types.SystemContext, registry string) (string, string, error) {
+func GetAuthentication(sys *types.SystemContext, registry string) (DockerAuthConfig, error) {
 	if sys != nil && sys.DockerAuthConfig != nil {
 		logrus.Debug("Returning credentials from DockerAuthConfig")
-		return sys.DockerAuthConfig.Username, sys.DockerAuthConfig.Password, nil
+		return DockerAuthConfig{
```

Comment:
It should be possible for a caller to specify an identity token via `SystemContext`.

So, maybe add the IdentityToken to `types.DockerAuthConfig` and don’t introduce the new `config.DockerAuthConfig` type. (Or keep the current `SystemContext.DockerAuthConfig` and add a separate `SytemContext.DockerIdentityToken` value, if they are always going to be exclusive.)

Reply:
Will do.

Reply:
Adding IdentityToken to `types.DockerAuthConfig`. These variables should be grouped together. It's either username + password or username + identity token.